GlebZhidovich/nodejs-aws-be
# Task 9 (Backend For Frontend)

Provide your reviewers with the following information:

task - https://github.com/EPAM-JS-Competency-center/cloud-development-course-initial/blob/main/task9-bff-elastic-beanstalk/task.md
link to the repo - https://github.com/GlebZhidovich/nodejs-aws-be/tree/task9
product-service service API endpoint URL - https://4g1i7l4cmg.execute-api.eu-west-1.amazonaws.com/dev/products
example of the create product API call with all needed information: URL, payload, headers, etc.
Swagger - https://app.swaggerhub.com/apis/GlebZhidovich/ProductService/1.0

CART service API endpoint URL - http://glebzhidovich-cart-api-dev.eu-west-1.elasticbeanstalk.com/api/profile/cart
bff-service service URL - http://glebzhidovich-bff-api-dev.eu-west-1.elasticbeanstalk.com/
example how to call product-service and CART services via bff-service service URL:
http://glebzhidovich-bff-api-dev.eu-west-1.elasticbeanstalk.com/cart
http://glebzhidovich-bff-api-dev.eu-west-1.elasticbeanstalk.com/products

3 - A working and correct express application should be in the bff-service folder. Reviewer can start this application locally with any valid configuration in the .env file and this application should works as described in the task 9.1
5 - The bff-service should be deployed with Elastic Beanstalk. The bff-service call should be redirected to the appropriate service : product-service or CART. The response from the bff-service should be the same as if product-service or CART services were called directly.

+1 - Add a cache at the bff-service level for a request to the getProductsList function of the product-service. The cache should expire in 2 minutes.
How to test:
Get products list
Create new product
Get products list - result shouldn’t have new product
Wait more than 2 minutes
Get products list - result should have new product
+1 - Use NestJS to create bff-service instead of express

I have done the whole task
